lib/tpm-v2.c

```diff
@@ -399,6 +399,23 @@ u32 tpm2_get_capability(struct udevice *dev, u32 capability, u32 property,
 	return 0;
 }
 
+u32 tpm2_algo_get_mask_from_hash(enum tpm2_algorithms hash)
+{
+	switch (hash) {
+	case TPM2_ALG_SHA1:
+		return TCG2_BOOT_HASH_ALG_SHA1;
+	case TPM2_ALG_SHA256:
+		return TCG2_BOOT_HASH_ALG_SHA256;
+	case TPM2_ALG_SHA384:
+		return TCG2_BOOT_HASH_ALG_SHA384;
+	case TPM2_ALG_SHA512:
+		return TCG2_BOOT_HASH_ALG_SHA512;
+	default:
+		break;
+	}
+	return 0;
+}
+
```

Comment:
Instead of adding a new function, do this

Change hash_algo_list and add another boolean 'enabled'. Set that to true or false if u-boot doesn't support it and use the existing functions